### PR TITLE
ADD new discord bot commands

### PR DESCRIPTION
**Pull request by LevBernstein**
Was unable to test the report command due to difficulties in setting up the database on my end. Other commands work fine.

Features to add:

- Consistent embed scheme: colors, author, thumbnail, etc
- Potentially enable discord.Intents if the bot should be able to interact with users on a more significant level
- Add setup guide for bot in README
- Add more commands!

### CODE REVIEW

File: api/jobs/bot.py [hunk 1]
```diff
@@ -75,69 +69,65 @@ def get_course_theia_session_count(course) -> int:
...
-    report += tabulate(data, headers=["Course Name", "Course Code", "Total IDEs opened"])
-
     return report
 
 
-bot = commands.Bot(command_prefix='!')
+bot = commands.Bot(
+    command_prefix="!",
+    case_insensitive=True,
+    help_command=None,
```
Comment (LevBernstein) [inline api/jobs/bot.py:124]: The options for a custom help command are either to create a minimal help command object, which is rather tedious, or set the help command to None and then just have a command named help; I did the latter here

File: api/jobs/bot.py [hunk 2]
```diff
@@ -45,9 +42,7 @@ def get_ides_opened_this_semester() -> int:
 
     :return: Total number of IDEs opened this semester
     """
-    return TheiaSession.query.join(
-        Course, TheiaSession.course_id == Course.id
-    ).count()
+    return TheiaSession.query.join(Course, TheiaSession.course_id == Course.id).count()
```
Comment (AlpacaMax) [inline api/jobs/bot.py:45]: I completely understand why you want to make this one line. The problem is that this line now has more than 80 characters, which is generally considered a bad coding style. If you check out other scripts in `api` you'll see similar statements or expressions which could be written in one line but are written in multiple lines just to follow this 80 characters rule. So please fix all the changes you made that breaks this rule. Thanks.

File: api/jobs/bot.py (hunk 1 above)
Comment (AlpacaMax) [reply api/jobs/bot.py:124]: Works pretty well. We can go with this.

File: api/jobs/bot.py [hunk 3]
```diff
@@ -75,69 +69,65 @@ def get_course_theia_session_count(course) -> int:
...
-
     return report
 
 
-bot = commands.Bot(command_prefix='!')
+bot = commands.Bot(
+    command_prefix="!",
+    case_insensitive=True,
+    help_command=None,
+    owner_id=448648593314807809,
```
Comment (AlpacaMax) [inline api/jobs/bot.py:125]: Please don't add your `owner_id` here! I just checked the documentation, looks like this parameter can be omitted. If you insist to leave this here, please make it into an environment variable as what I did for the bot token.

File: api/jobs/bot.py [hunk 4]
```diff
@@ -146,5 +136,32 @@ async def report(ctx):
...
+@bot.command(name="contribute", aliases=("github",), help="Contributing to Anubis")
+async def contribute(ctx, *args):
+    """
+    Respond to `!contribute` command with a link to the GitHub repo
+
+    :return:
+    """
+    emb = discord.Embed(
+        title="Contributing to Anubis",
+        description="Thanks for your interest in contributing to Anubis! We're always looking for new people to help make Anubis even better. Please head to [our GitHub repo](https://github.com/anubislms/Anubis).",
```
Comment (AlpacaMax) [inline api/jobs/bot.py:148]: This line is way too long. You can do something like:
``` python
description ="Thanks for your interest in contributing to Anubis! "
description += "We're always looking for new people to help make"
description += "Anubis even better ..."
...
```

**Review by AlpacaMax — CHANGES_REQUESTED** [review]
Good stuff! There are some small issues with these changes. Once you fix these we can merge it to master.

File: api/jobs/bot.py (hunk 2 above)
Comment (LevBernstein) [reply api/jobs/bot.py:45]: I didn't do so; I ran black on the file for linting, which formatted it like that.

Comment (wabscale) [reply api/jobs/bot.py:45]: We've been using 120 as the max line length. So there is no problem with this line

File: api/jobs/bot.py (hunk 3 above)
Comment (wabscale) [reply api/jobs/bot.py:125]: If it doesnt need to be there, we should probably avoid magic numbers

**Review by wabscale — APPROVED** [review]

**Comment by wabscale** [comment]
I did a squash & rebase manually for this. It is in master.

**Pull request closed without merging**